### PR TITLE
Diagnostics => DiagnosticSet

### PR DESCRIPTION
**Pull request by dabrahams**
Diagnostics is just a notional set, so let's make its usage, API, and documentation more consistent with Set.

### CODE REVIEW

File: Sources/Core/DiagnosticSet.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import Foundation
```
Comment (kyouko-taiga) [inline Sources/Core/DiagnosticSet.swift:1]: I'm surprised to see Foundation here.

Comment (dabrahams) [reply Sources/Core/DiagnosticSet.swift:1]: Me TOO!

Comment (dabrahams) [reply Sources/Core/DiagnosticSet.swift:1]: I know why it's there: it used to be writing to stderr.

**Review by kyouko-taiga — APPROVED** [review]

File: Sources/Core/DiagnosticSet.swift (hunk 1 above)
Comment (kyouko-taiga) [reply Sources/Core/DiagnosticSet.swift]: Thought so. Good opportunity to get rid of this import. Everything else looks good.

Comment (dabrahams) [reply Sources/Core/DiagnosticSet.swift]: removed.

**Pull request merged**